lib/types/array.js

```diff
@@ -661,19 +661,29 @@ internals.fillMissedErrors = function (schema, errors, requireds, value, state,
 };
 
 
-internals.fillOrderedErrors = function (schema, errors, ordereds, value, state, prefs) {
+internals.checkRemainingOrdereds = function (schema, errors, ordereds, value, state, prefs) {
 
     const requiredOrdereds = [];
+    const orderedsWithDefault = [];
 
     for (const ordered of ordereds) {
         if (ordered._flags.presence === 'required') {
             requiredOrdereds.push(ordered);
```

Comment:
Almost forgot, you need a `continue` statement here

Reply:
```suggestion
            requiredOrdereds.push(ordered);
            continue;
```

Reply:
But as I said, it would make more sense to fill default values outside of this loop, because it wouldn't make sense to invoke `.validate()` so many times when errors are going to be thrown anyway.

Reply:
Yes, i forgot it, committed. 

> But as I said, it would make more sense to fill default values outside of this loop, because it would be really expensive to invoke .validate() so many times when errors are going to be thrown anyway.

wouldn't be more expensive if we made a separate for loop for this?

Reply:
@iifawzi Loops are really, really fast. Much faster than running multiple `$_validate()`s. Having a separate loop for filling errors would yield much better performance since Joi can stop performing these expensive validations if some required schemas are not provided. Think about it this way. If you have an array of 99 schemas with default values, and only 1 required schema, finding this 1 required schema and providing the error would be much faster and efficient than looping through 100 items, running `$_validate()` 99 times only to return the exact same error after.

Reply:
It might not quite make sense at first because if you were to have 2 loops, you would have to iterate through 200 items. However, the first iteration would be really fast because all we are doing is just extracting labels.